risc0/circuit/recursion/build.rs

```diff
@@ -50,15 +52,25 @@ fn download_zkr() {
         return;
     }
 
-    if out_path.exists() {
-        if check_sha2(&out_path) {
+    // Check the ZKR archive in the out dir and the src dir. Use the out file if it matches the
+    // hash above, but also warn the developer if there exists a copy in the src dir and it does
+    // not match.
+    let zkr_out_valid = out_path.exists() && check_sha2(&out_path);
+
+    if src_path.exists() {
+        if check_sha2(&src_path) {
+            if !zkr_out_valid {
+                fs::copy(&src_path, &out_path).unwrap();
+            }
```

Comment:
I think we need to be careful about changes here. The last set of changes were specifically allow nix users to be able to override this behavior so that they can ensure no downloads occur during a build. What specifically was the goal with this refactor?

Reply:
My goal here is to get a warning if I forget to update the hash in the `build.rs` file above. It's something I had forgotten needed to be done, and took me a minute to remember we I first started, and is also a step that's easy to forget when making multiple iterations.

Reply:
I'm trying to remember why we had a `remove_file` that is now dropped, so it does feel like something functional did change, but I don't fully understand this diff.

Reply:
I've reverted this change in 9df10c28d647e1b7ce480d048601e7aea2bb691b